internal/storage/fs/object/blob/blob_fs_test.go

```diff
@@ -52,7 +52,7 @@ func Test_FS(t *testing.T) {
 		}
 		require.NoError(t, bucket.Close())
 
-		azfs, err := NewFS(logger, "file", dir)
+		azfs, err := NewFS(ctx, StrUrl(fileblob.Scheme, dir), dir, "data")
```

Comment:
I think this is all that is needed @erka 👍 Now the NewFS prefix needs to match the prefix used for inserting entries into the bucket.
```suggestion
		azfs, err := NewFS(ctx, StrUrl(fileblob.Scheme, dir), dir, "data/")
```

Reply:
Side note: while walking the blob code, I only just realize that `*blob.Bucket` already implements `io/fs`. I'm not sure, but just returning the instance from `OpenBucket` might just work out of the box? (probably won't, but didn't initially realize they already implemented all the `io/fs` stuff on that type).

Reply:
I had a quick play with ☝️ just using it as-is and its not quite the right fit just out of the box. But making the change I suggested above gets the unit tests passing 👍

Reply:
We could do that but I suspect that it will affect existing users. 

> but the config just used a-prefix-name without slashes, 
https://github.com/flipt-io/flipt/pull/2589#discussion_r1438670982

Probably I need to add one more integration test for s3 with prefix.

Reply:
If it isn't too painful to add, an IT with a prefix would be great. But I think now that we're not forcible adding the slash to the prefixes it won't break their usecase. They just threaded the prefix through without slashes before:
https://github.com/flipt-io/flipt/blob/77e21fd62a00c6d2d4fd55a7501e6a8a95404e2e/internal/storage/fs/object/s3/s3fs.go#L71-L81

Reply:
The reason this failed without putting in the `/` at the end, was because the test expectation is that the resulting files are named `"one"` and `"two"`, but when you supply a prefix of just `data`, they become `"/one"` and `"/two"` in the expected map.

Reply:
I might be missing something as to why providing just `data` doesn't work here.

Reply:
You are correct. 

After playing with it a bit I understand that `something.flipt.yml` could be on S3 and using `something` as a prefix will work. If I introduce the flash it will fail. I just mixed `folder` and `prefix` into one thing.